x-pack/plugins/session_view/public/components/ProcessTreeNode/index.tsx

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ *2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo, useRef, useLayoutEffect, useState, useEffect, MouseEvent } from 'react';
+import { EuiButton, EuiIcon, EuiToolTip } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { Process } from '../../../common/types/process_tree';
+import { useStyles, ButtonType } from './styles';
+import { ProcessTreeAlerts } from '../ProcessTreeAlerts';
+
+interface ProcessDeps {
+  process: Process;
+  isSessionLeader?: boolean;
+  depth?: number;
+  onProcessSelected?: (process: Process) => void;
+}
+
+/**
+ * Renders a node on the process tree
+ * TODO: as well as sections for tty output, alerts and file redirection.
+ */
+export function ProcessTreeNode({
+  process,
+  isSessionLeader = false,
+  depth = 0,
+  onProcessSelected,
+}: ProcessDeps) {
+  const textRef = useRef<HTMLSpanElement>(null);
+
+  const [childrenExpanded, setChildrenExpanded] = useState(isSessionLeader || process.autoExpand);
+  const [alertsExpanded, setAlertsExpanded] = useState(false);
+  const [showGroupLeadersOnly, setShowGroupLeadersOnly] = useState(isSessionLeader);
+  const { searchMatched } = process;
+
+  useEffect(() => {
+    setChildrenExpanded(isSessionLeader || process.autoExpand);
+  }, [isSessionLeader, process.autoExpand]);
+
+  const processDetails = useMemo(() => {
+    return process.getDetails();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can you remove the `eslint-disable-next-line` and add the `process.getDetails` reference? The reference shouldn't change right? I think you can still add `process.events.length` as long as the getDetails reference is here as well. We've run into weird bugs disabling dependency lists in the past and if it is avoidable that'll be great, but if there is no workaround, no worries.

Reply:
It will actually complain about what it thinks is an unnecessary dependency. I will add process.getDetails, but I'll need to keep the eslint rule there so it doesn't complain about process.events.length. Would be nice to move this memo to the class function itself, but not too sure if you can memoize a class function by a class property (the events array in this case).

Reply:
I may try and create an intermediate memoized function for getDetails, getAlerts etc that uses this.events as a parameter to break it's cache. If that works I can remove the useMemo in the react component.

Reply:
was able to make use of memoizeOne for a few functions in ProcessImpl (see ProcessTree/hooks.ts). 

Will mark resolved for now. plz reopen if you find problems with this approach.